pg_auth_mon.c

```diff
@@ -191,6 +194,15 @@ auth_monitor(Port *port, int status)
 		fai->key = key;
 		memset(&fai->total_successful_attempts, 0, sizeof(auth_mon_rec)
 			   - offsetof(auth_mon_rec, total_successful_attempts));
+		/*
+		 * We use the key equal to zero to aggregate login attempts
+		 * of non-existing users. For them it does makes no sense to
+		 * persist any particular username, so we leave initial_rolename
+		 * blank.
+	 	 */
+		if (key != 0) {
```

Comment:
instead of 0, you may use InvalidOid, since get_role_oid will be returning that in case of not existing users.

Reply:
good point , will fix